include/ozo/result.h

```diff
@@ -391,13 +391,40 @@ class basic_result {
      * @return native_handle_type --- native handle representation
      */
     native_handle_type native_handle() const noexcept {
-        return std::addressof(*res_);
+        return std::addressof(*handle());
     }
 
+    /**
+     * Checks if object contains result handle.
+     */
+    bool valid() const noexcept { return !ozo::is_null(handle()); }
+
+protected:
+    handle_type& handle() { return res_;}
+    const handle_type& handle() const { return res_;}
+
 private:
     handle_type res_;
 };
 
+/**
+ * @brief Database raw result representation
+ *
+ * Copyable version of `ozo::result`. The result object is useful then it needs to get an access
+ * to raw data representation or the underlying `libpq` handle.
+ *
+ * Get this type of result via `share()` method call:
+ *
+ * @code
+ozo::result res;
+//...
+share_result sres = res.share();
+ * @endcode
+ *
+ * @ingroup group-requests-types
+ */
+using shared_result = basic_result<pg::shared_safe_handle_t<pg::result>>;
```

Comment:
Should we add tests for a new API type?

Reply:
Well, we already have [ozo::basic_result tests](https://github.com/yandex/ozo/blob/master/tests/result.cpp#L184) that cover all the functionality but the integration tests certainly should be added. Thanks for the notice!